### ISSUE TITLE
Throw an error on Infinity of NaN values

### ISSUE THREAD

**Issue by michael-markl** [erdtman/canonicalize#11]
According to the [spec](https://www.rfc-editor.org/rfc/rfc8785#section-3.2.2.3-3):

>  Note: Since Not a Number (NaN) and Infinity are not permitted in JSON, occurrences of NaN or Infinity MUST cause a compliant JCS implementation to terminate with an appropriate error.

I believe, this is currently not the case. Instead, `JSON.canonicalize(Number.POSITIVE_INFINITY)` currently returns `null`.

**Comment by erdtman** [comment]
Thanks for noticing, I will think about a good approach to get an error for these cases. I'm open for suggestions if you have any.

**Comment by michael-markl** [comment]
One approach I could think of would be to prepend your algorithm with:
```js
  // Throw on NaN or infinite numbers (https://www.rfc-editor.org/rfc/rfc8785#section-3.2.2.3-3)
  if (typeof object === 'number' && (isNaN(object) || !isFinite(object)) {
      throw new Error("Canonocalize does not support NaN or infinite numbers.")
  }
```

I guess, this should be marked as a breaking change in the Changelog (and version number if you're doing semantic versioning)

**Comment by erdtman** [comment]
Thanks, did a change similar to what you suggested.

Please have a look https://github.com/erdtman/canonicalize/commit/7a3d6245d14aa9154d1cb4247c5236630373c8b8

If it looks sane to you as well, I'll publish an update

**Comment by michael-markl** [comment]
LGTM! Thanks for the update!

**Comment by michael-markl** [comment]
Fixed in 2.0.0
